python_src/program_markers/instrumenter.py

```diff
@@ -633,6 +700,48 @@ def remove_markers(old_markers: Sequence[Marker]) -> tuple[Marker, ...]:
             vr_marker_defined_constants=preserved_vr_marker_constants,
         )
 
+    @staticmethod
+    def generate_macro(marker_strategy: MarkerStrategy, marker: DCEMarker) -> str:
+        """Returns the preprocessor macro text for the given marker.
+
+        Args:
+            marker_strategy (MarkerStrategy):
+                the marker strategy to be used in the macro
+            marker (DCEMarker):
+                the marker for which a macro should be made
+
+        Returns:
+            str:
+                a string containing the preprocessor macro text
+        """
+        id = marker.marker[len(marker.prefix) : -1]
+        macro_name = f"DCEMARKERMACRO{id}_"
```

Comment:
Instead of hardcoding this here, can we add a method in `DCEMarker`?

Reply:
Of course, we can even use the already existing `macro` method, which I had forgotten about. Then there would only be one line left in the `generate_macro` method and since it is only used once in the `with_marker_directives` method I propose that we just inline it.